ci-operator/step-registry/cucushift/installer/check/vsphere/cpms/cucushift-installer-check-vsphere-cpms-commands.sh

```diff
@@ -0,0 +1,91 @@
+    else
+	echo "ERROR: The failure domain name are different between infrastructure and cmps. please check"
+        check_result=1
+    fi
+    return ${check_result}
+}
+
+
+function checkSinglezone() {
+        check_result=0
+	node_name=$(oc get machines.machine.openshift.io -n openshift-machine-api --selector machine.openshift.io/cluster-api-machine-type=master --no-headers | awk '{print $1}' | head -n1)
+	machine_workspace=$(oc get machine ${node_name} -n openshift-machine-api -o json | jq -r '.spec.providerSpec.value.workspace')
+	machine_template=$(oc get machine ${node_name} -n openshift-machine-api -o json | jq -r '.spec.providerSpec.value.template')
+	machine_network=$(oc get machine ${node_name} -n openshift-machine-api -o json | jq -r '.spec.providerSpec.value.network.devices[].networkName')
+        
+	cpms_network=$(oc get controlplanemachineset -n openshift-machine-api -o json | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.network.devices[].networkName')
+	cpms_workspace=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.workspace')
+        cpms_template=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.template')
+        if [[ ${machine_workspace} == "${cpms_workspace}" ]] && [[ ${machine_template} == "${cpms_template}" ]]  && [[ ${machine_network} == "${cpms_network}" ]]; then
+	    echo "INFO: The network,workspace and template setting between cmps and machines are same. that's expected"
+	else
+	   check_result=1
+	fi
+    return ${check_result}
+}
+
+function checkTemplate() {
+
+#check template in infrastructure should be populated.
+    check_result=0
+    infra_template=$(oc get infrastructure -n openshift-machine-api -ojson | jq -r '.items[].spec.platformSpec.vsphere.failureDomains[].topology.template')
+    if [[ -n ${infra_template} ]]; then
+        echo "INFO: The template is populated. template under infrastructure is ${infra_template}"
+    else
+        echo "ERROR: The template is empty, please check"
+        check_result=1
+    fi
+    return ${check_result}
+
+}
+
+ocp_minor_version=$(oc version -o json | jq -r '.openshiftVersion' | cut -d '.' -f2)
+
+if (( ${ocp_minor_version} < 15 )); then
+	echo "CPMS failureDomain check is only available on 4.15+ cluster, skip the check!"
+        exit 0
+elif (( ${ocp_minor_version} >= 15 )) && [[ "${FEATURE_SET}" == "TechPreviewNoUpgrade" ]]; then
+      failuredomain_count=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.failureDomains.vsphere[].name' |wc -l)
+      if (( ${failuredomain_count} > 1 )); then
+	    if  checkMultizone && checkTemplate; then
+		    exit 0
```

Comment:
Because in this script, `set -o errexit` is enabled, so any function failure would lead the whole script exit as failure, so do not need a explicit exit.
Suggesting:
```
      checkTemplate
      failuredomain_count=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.failureDomains.vsphere[].name' |wc -l)
      if (( ${failuredomain_count} > 1 )); then
          checkMultizone
      elif (( ${failuredomain_count} = 1 )); then
          checkSinglezone
      fi
```

Reply:
ok, understand.